examples/bsc-p2p/src/block_import/service.rs

```diff
@@ -0,0 +1,266 @@
+
+        (
+            Self {
+                engine,
+                provider,
+                from_network,
+                to_network,
+                pending_imports: FuturesUnordered::new(),
+            },
+            ImportHandle::new(to_import, import_outcome),
+        )
+    }
+
+    /// Sends a forkchoice update to the engine with the given block hash and number
+    async fn send_fork_choice_update(
+        engine: &BeaconConsensusEngineHandle<T>,
+        provider: &Provider,
+        block_hash: B256,
+        block_number: BlockNumber,
+    ) -> Result<(), BlockImportError> {
+        let current_head =
+            provider.best_block_number().map_err(|e| BlockImportError::Other(e.into()))?;
+        let current_head_hash = provider
+            .block_hash(current_head)
+            .map_err(|e| BlockImportError::Other(e.into()))?
+            .ok_or_else(|| BlockImportError::Other("Current head hash not found".into()))?;
+
+        let head_block_hash =
+            canonical_head(block_hash, block_number, current_head, current_head_hash);
+
+        let state = ForkchoiceState {
+            head_block_hash,
+            safe_block_hash: current_head_hash,
+            finalized_block_hash: current_head_hash,
+        };
+
+        engine
+            .fork_choice_updated(state, None, EngineApiMessageVersion::default())
+            .await
+            .map_err(|e| BlockImportError::Other(e.into()))
+            .and_then(|response| match response.payload_status.status {
+                PayloadStatusEnum::Valid => Ok(()),
+                PayloadStatusEnum::Invalid { validation_error } => {
+                    Err(BlockImportError::Other(validation_error.into()))
+                }
+                PayloadStatusEnum::Syncing => {
+                    Err(BlockImportError::Other(PayloadStatusEnum::Syncing.as_str().into()))
+                }
+                _ => Err(BlockImportError::Other("Unsupported forkchoice payload status".into())),
+            })
+    }
+
+    /// Sends a new payload to the engine and waits for the outcome.
+    /// If it is valid, proceeds with Forkchoice Update (FCU).
+    fn import(&self, block: BlockMsg<T>, peer_id: PeerId) -> ImportFuture<T> {
+        let engine = self.engine.clone();
+        let provider = self.provider.clone();
+
+        Box::pin(async move {
+            let sealed_block = block.block.block.clone().seal();
+            let hash = sealed_block.hash();
+            let number = sealed_block.number();
+            let payload = T::block_to_payload(sealed_block);
+
+            match engine.new_payload(payload).await {
+                Ok(payload_status) => match payload_status.status {
+                    PayloadStatusEnum::Valid => {
+                        // Send forkchoice update if the payload is valid
+                        if let Err(err) =
+                            Self::send_fork_choice_update(&engine, &provider, hash, number).await
```

Comment:
let's do this in two steps and avoid including to much in a single operation

new payload -> 1 future
on outcome -> next future